test/service-object.ts

```diff
@@ -342,15 +339,10 @@ describe('ServiceObject', () => {
     it('should execute callback with error if not 404', (done) => {
       const error = new ApiError('');
       error.code = 500;
-
-      serviceObject.get =
-          (configOrCallback: SO.GetConfig|SO.InstanceResponseCallback,
-           callback?: SO.InstanceResponseCallback) => {
-            callback = typeof configOrCallback === 'function' ?
-                configOrCallback :
-                callback;
-            callback!(error);
-          };
+      // tslint:disable-next-line:no-any
+      (serviceObject as any).get = (callback: SO.InstanceResponseCallback) => {
+        callback(error);
```

Comment:
It seemed to me that the previous code that mocked `serviceObject.get` was unnecessarily complex with the goal of being typed consistently with `serviceObject.prototype.get`. These "true-to-prototype" typings in the unit tests, however, failed to reveal that those typings were in fact incorrect. This PR therefore simplifies the mocked `serviceObject.get` definitions in the unit tests, preserving the runtime behavior but using the `any` escape valve for typings. I could be convinced (and already have some code toward this end on a branch) that instead we should abstract out all these `(serviceObject as any).get = ...` assignments into a lil helper function `mockGet` that assigns `serviceObject.get` in an appropriately-typed way.